polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -3157,36 +2893,33 @@ mlir::FunctionOpInterface MLIRASTConsumer::GetOrCreateMLIRFunction(
     llvm_unreachable("Invalid function context");
   };
 
+  // Create the MLIR function and set its visibility and function attributes.
   FunctionOpInterface function = createFunction(F);
+  LLVM::Linkage lnk = getMLIRLinkage(getLLVMLinkageType(*FD, ShouldEmit));
+  setMLIRFunctionVisibility(function, *FD, ShouldEmit);
+  setMLIRFunctionAttributes(function, F, lnk, module->getContext());
+  checkFunctionParent(function, F.getContext(), module);
 
-  if (LV == llvm::GlobalValue::InternalLinkage ||
-      LV == llvm::GlobalValue::PrivateLinkage || !FD->isDefined() ||
-      FD->hasAttr<CUDAGlobalAttr>() || FD->hasAttr<CUDADeviceAttr>() ||
-      !ShouldEmit) {
-    SymbolTable::setSymbolVisibility(function,
-                                     SymbolTable::Visibility::Private);
-  } else {
-    SymbolTable::setSymbolVisibility(function, SymbolTable::Visibility::Public);
-  }
-
-  setMLIRFunctionAttributes(function, F, lnk, builder.getContext());
+  // Decide whether the MLIR function should be emitted.
+  const FunctionDecl *Def = nullptr;
+  if (!FD->isDefined(Def, /*checkforfriend*/ true))
+    Def = FD;
 
   if (Def->isThisDeclarationADefinition()) {
     assert(Def->getTemplatedKind() !=
-           FunctionDecl::TemplatedKind::TK_FunctionTemplate);
-    assert(Def->getTemplatedKind() !=
-           FunctionDecl::TemplatedKind::
-               TK_DependentFunctionTemplateSpecialization);
+               FunctionDecl::TemplatedKind::TK_FunctionTemplate &&
+           Def->getTemplatedKind() !=
+               FunctionDecl::TemplatedKind::
+                   TK_DependentFunctionTemplateSpecialization);
     if (ShouldEmit) {
       LLVM_DEBUG(llvm::dbgs()
                  << __LINE__ << ": Pushing " << F.getContext() << " function "
                  << Def->getNameAsString() << " to functionsToEmit\n");
       functionsToEmit.emplace_back(*Def, F.getContext());
     }
-  } else if (ShouldEmit) {
-    emitIfFound.insert(name);
-  }
-  checkFunctionParent(function, F.getContext(), module);
+  } else if (ShouldEmit)
+    emitIfFound.insert(mangledName);
+
```

Comment:
II'd say removing braces hurts readability here as we have a previous branch with braces alread.